File: contest-2/c++/p14-Armstrong_Number.cpp
// write a program to check whether the given input is armstrong or not.

#include <cmath>
#include <cstdio>
#include <vector>
#include <iostream>
#include <algorithm>
using namespace std;


int main() {
    /* Enter your code here. Read input from STDIN. Print output to STDOUT */   
    int n,i,j,k;
    string s;
    cin>>s;
    int c=s.length();
    i=0;
    j=stoi(s);
    while(j)
    {
    k=j%10;
        k=pow(k,c);
        i+=k;
        j/=10;
    }
    if(i==stoi(s))
        cout<<"yes";
    else
        cout<<"no";
    
    
    
    
    return 0;
}

File: contest-2/python/p14-Armstrong_Number.py
# write a program to check whether the given input is armstrong or not.

# Note : A number is called armstrong number, if the sum of its own digits raised to the power number of digits gives the number itself .

s=input()
l=len(s)
x=0
for c in s:
    i=int(c)
    x+=i**l
y=int(s)
if(x==y):
    print("yes")
else:
    print("no")
